src/Data/Vec/Properties.agda

```diff
@@ -154,6 +159,70 @@ take≡truncate : ∀ m (xs : Vec A (m + n)) →
 take≡truncate zero    _        = refl
 take≡truncate (suc m) (x ∷ xs) = cong (x ∷_) (take≡truncate m xs)
 
+truncate-zipWith : (f : A → B → C) (m≤n : m ≤ n) (xs : Vec A n) (ys : Vec B n) →
+  truncate m≤n (zipWith f xs ys) ≡ zipWith f (truncate m≤n xs) (truncate m≤n ys)
+truncate-zipWith f z≤n xs ys = refl
+truncate-zipWith f (s≤s m≤n) (x ∷ xs) (y ∷ ys) =
+  cong (f x y ∷_) (truncate-zipWith f m≤n xs ys)
+
+truncate-zipWith-truncate : ∀ (f : A → B → C) (o≤m : o ≤ m) (m≤n : m ≤ n)
+  (xs : Vec A n) (ys : Vec B m) →
+  let o≤n = ≤-trans o≤m m≤n in
+  truncate o≤m (zipWith f (truncate m≤n xs) ys) ≡
+  zipWith f (truncate o≤n xs) (truncate o≤m ys)
+truncate-zipWith-truncate f o≤m m≤n xs ys =
+  let o≤n = ≤-trans o≤m m≤n in
+  trans (truncate-zipWith f o≤m (truncate m≤n xs) ys)
+  (cong (λ zs → zipWith f zs (truncate o≤m ys)) ((sym (truncate-trans o≤m m≤n xs))))
+
+zipWith-truncate : ∀ (f : A → B → C) {p q : ℕ} (xs : Vec A (p + q)) (ys : Vec B (p + q)) →
+  let p≤p+q = m≤m+n p q in
+  zipWith f (truncate p≤p+q xs) (truncate p≤p+q ys) ≡ truncate p≤p+q (zipWith f xs ys)
+zipWith-truncate f {p} {q} xs ys =
+  let p≤p+q = m≤m+n p q in
+  begin
+    zipWith f (truncate p≤p+q xs) (truncate p≤p+q ys) ≡⟨ cong₂ (zipWith f) (take≡truncate p xs) (take≡truncate p ys) ⟨
+    zipWith f (take p xs) (take p ys)                 ≡⟨ take-zipWith f xs ys ⟨
+    take p (zipWith f xs ys)                          ≡⟨ take≡truncate p (zipWith f xs ys) ⟩
+    truncate p≤p+q (zipWith f xs ys) ∎ where open ≡-Reasoning
```

Comment:
Assuming this meets the Fairbairn threshold, it's still very hard to parse, but AFAICT, this is a specialised instance of `truncate-zipWith` above... or am I missing something?
Suggest: deletion